### ISSUE TITLE
[a11y]: ai-chat does not sufficiently set aria regions/landmarks to indicate what is expanded/collapsed with a linkList element

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#911]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

Consider the collapsable/expandable section of a linkList element. Observed Missing Regions /ARIA Land marks unable to identify the content boundaries of the Expand/collapse section.

Recommended fix: For the benefits of Screen reader users , Define the Expand/collapse content with in a Labeled Region , Label should be descriptive to indicate the content with in the Region.

User impact: Impact screen reader users: When content sections are not enclosed within appropriately labeled regions, screen reader users face barriers in understanding
the boundaries of each content section, and also struggle to understand the overall page structure, leading to increased cognitive load and navigation
frustration.

### WCAG 2.1 Violation

WCAG Mapping: 1.3.1 Info and Relationships

### Reproduction/example

n/a

### Steps to reproduce

1. open ai-chat
2. ask a question that is answered with a list of references/links, i.e. a linkList element
3. observer insufficient region/landmark definitions

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
